### PR TITLE
Fixes stylecop warnings on FolderManager.cs

### PR DESCRIPTION
**Pull request by valadas**
@david-poindexter @mitchelsellers I fixed the stylcop issues on the FolderManager.cs class for fixing the search issues after with a cleaner PR that does not have huge changes.

### CODE REVIEW

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 1]
```diff
@@ -438,6 +447,11 @@ public virtual IEnumerable<IFolderInfo> GetFolders(UserInfo user, string permiss
             return userFolders;
         }
 
+        /// <summary>
+        /// Gets the folder that belongs to a specific user.
+        /// </summary>
+        /// <param name="userInfo">The user to get the folder for.</param>
+        /// <returns>The informaiton about the the user folder, <see cref="IFolderInfo"></see>.</returns>
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:454]:
```suggestion
        /// <returns>The information about the the user folder, <see cref="IFolderInfo"></see>.</returns>
```

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 2]
```diff
@@ -838,23 +862,41 @@ public virtual IFolderInfo MoveFolder(IFolderInfo folder, string newFolderPath)
...
 
+        /// <summary>
+        /// Adds a log entry.
+        /// </summary>
+        /// <param name="folder">The folder to log about.</param>
+        /// <param name="eventLogType">The type of the log entry.</param>
         internal virtual void AddLogEntry(IFolderInfo folder, EventLogController.EventLogType eventLogType)
         {
-            EventLogController.Instance.AddLog(folder, PortalController.Instance.GetCurrentPortalSettings(), this.GetCurrentUserId(), string.Empty, eventLogType);
+            EventLogController.Instance.AddLog(folder, (PortalSettings)PortalController.Instance.GetCurrentSettings(), this.GetCurrentUserId(), string.Empty, eventLogType);
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:883]: Do we need to add an `IPortalSettings` overload for `AddLog`?

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 3]
```diff
@@ -838,23 +862,41 @@ public virtual IFolderInfo MoveFolder(IFolderInfo folder, string newFolderPath)
...
+        /// <param name="propertyValue">The value of the property.</param>
+        /// <param name="eventLogType">The type of log entry.</param>
         internal virtual void AddLogEntry(string propertyName, string propertyValue, EventLogController.EventLogType eventLogType)
         {
-            EventLogController.Instance.AddLog(propertyName, propertyValue, PortalController.Instance.GetCurrentPortalSettings(), this.GetCurrentUserId(), eventLogType);
+            EventLogController.Instance.AddLog(propertyName, propertyValue, (PortalSettings)PortalController.Instance.GetCurrentSettings(), this.GetCurrentUserId(), eventLogType);
         }
 
         /// <summary>This member is reserved for internal use and is not intended to be used directly from your code.</summary>
+        /// <param name="portalId">The site (portal) ID.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:898]: 👍

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 4]
```diff
@@ -1065,6 +1129,8 @@ internal virtual void DeleteFolder(int portalId, string folderPath)
         }
 
         /// <summary>This member is reserved for internal use and is not intended to be used directly from your code.</summary>
+        /// <param name="folderMappings">The folder mapping affected.</param>
+        /// <param name="foldersToDelete">A list of folders to delete.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:1133]:
```suggestion
        /// <param name="foldersToDelete">The folders to delete.</param>
```

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 5]
```diff
@@ -2292,25 +2434,27 @@ public int Compare(string x, string y)
...
+            /// <param name="source">The source folder.</param>
+            /// <param name="target">The destination folder.</param>
             public MoveFoldersInfo(string source, string target)
             {
                 this.Source = source;
                 this.Target = target;
             }
 
+            /// <summary>
+            /// Gets the Souce folder.
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:2450]:
```suggestion
            /// Gets the Source folder.

```

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs (hunk 2 above)
Comment (valadas) [reply DNN Platform/Library/Services/FileSystem/FolderManager.cs]: Well, yeah, I guess, now I was not sure how we should properly do it once there, do we have IPortalSettings for DI and if so do we inject it in the EventLogController. That was the quick solution to just fix the StyleCop warnings, but I agree we should not have to cast it. Any advice ?

**Comment by valadas** [comment]
@bdukes I see multiple of your suggestions with the "commit suggestion" button greyed out, that means you did it?

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 6]
```diff
@@ -2,2315 +2,2460 @@
...
+            var portalId = user.IsSuperUser ? Null.NullInteger : user.PortalID;
+
+            var folderMapping = FolderMappingsConfigController.Instance.GetFolderMapping(portalId, DefaultUsersFoldersPath) ?? FolderMappingController.Instance.GetDefaultFolderMapping(portalId);
+
+            if (!this.FolderExists(portalId, DefaultUsersFoldersPath))
+            {
+                this.AddFolder(folderMapping, DefaultUsersFoldersPath);
+            }
+
+            // GetUserFolderPathElement is depreced without a replacement, it should have been internal only and will be removed in Dnn10, hence the warning disable here
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:928]:
```suggestion
            // GetUserFolderPathElement is deprecated without a replacement, it should have been internal only and will be removed in DNN v10, hence the warning disable here
```

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 7]
```diff
@@ -2,2315 +2,2460 @@
...
+                foreach (var fileId in fileIds)
+                {
+                    DataCache.RemoveCache("GetFileById" + fileId);
+                }
+            }
+        }
+
+        /// <summary>This member is reserved for internal use and is not intended to be used directly from your code.</summary>
+        /// <returns>The added folder information, <see cref="IFolderInfo" />.</returns>
+        /// <param name="user">the user to add a forder for.</param>
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:915]:
```suggestion
        /// <param name="user">the user to add a folder for.</param>
```

File: DNN Platform/Library/Services/FileSystem/FolderManager.cs [hunk 8]
```diff
@@ -2,2315 +2,2460 @@
...
+#pragma warning restore 612,618
+
+            var folderPath = PathUtils.Instance.FormatFolderPath(string.Format(DefaultUsersFoldersPath + "/{0}", rootFolder));
+
+            if (!this.FolderExists(portalId, folderPath))
+            {
+                this.AddFolder(folderMapping, folderPath);
+            }
+
+            // GetUserFolderPathElement is depreced without a replacement, it should have been internal only and will be removed in Dnn10, hence the warning disable here
```
Comment (bdukes) [inline DNN Platform/Library/Services/FileSystem/FolderManager.cs:940]:
```suggestion
            // GetUserFolderPathElement is deprecated without a replacement, it should have been internal only and will be removed in DNN v10, hence the warning disable here
```

**Comment by bdukes** [comment]
I haven't committed any of those changes, but maybe @mitchelsellers' commit included some of them.

**Comment by valadas** [comment]
Ok, so I accepted the other suggestions, we just have to decide on the cast or adding an overload...

**Comment by bdukes** [comment]
I just pushed a change to add overloads (and deprecate the old overloads).

**Comment by valadas** [comment]
Awesome @bdukes Love it.

**Pull request merged**